lapce-app/src/app.rs

```diff
@@ -3719,3 +3730,70 @@ pub fn window_menu(
                 })),
         )
 }
+
+fn panic_hook() {
+    std::panic::set_hook(Box::new(move |info| {
+        let thread = std::thread::current();
+        let thread = thread.name().unwrap_or("main");
+        let backtrace = backtrace::Backtrace::new();
+
+        let payload = if let Some(s) = info.payload().downcast_ref::<&str>() {
+            s
+        } else {
+            "<unknown>"
+        };
+
+        match info.location() {
+            Some(loc) => {
+                error!(
+                    target: "lapce_app::panic_hook",
+                    "thread {thread} panicked at {} | file://./{}:{}:{}\n{:?}",
+                    payload,
+                    loc.file(), loc.line(), loc.column(),
+                    backtrace,
+                );
+            }
+            None => {
+                error!(
+                    target: "lapce_app::panic_hook",
+                    "thread {thread} panicked at {}",
+                    payload,
+                );
+            }
+        }
+
+        #[cfg(windows)]
+        error_modal("Error", &info.to_string());
+    }))
+}
+
+#[cfg(windows)]
+fn error_modal(title: &str, msg: &str) -> i32 {
```

Comment:
I don't see the need to use a modal for panic messages.

Reply:
The log should be enough I think?

Reply:
Log doesn't give a visual clue that program failed to launch or why

Reply:
It would be more useful if mac and linux can have it so that it's more consistent between platforms.

Reply:
That's my goal but I don't know how to do that on mac/linux yet
macOS does have own pop-up though so I'm also unsure if that's needed, I'll need to test how it behaves
![app-quit-unexpectedly-crash-report-dialog-window-mac-610x405-3458731267](https://github.com/lapce/lapce/assets/50457605/d4dc5fba-a5cc-4ae8-85e4-75f96ef28871)
linux might be tricky because of wayland/xorg

Reply:
It makes sense to me now. We can add support for other platforms in other PRs.